src/components/NewCommentForm.tsx

```
@@ -1,8 +1,72 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Post } from '../types/Post';
+
+type Props = {
+  openedPost: Post;
+  isLoading: boolean;
+  addCommentHandler: (
+    postId: number,
+    name: string,
+    email: string,
+    body: string,
+  ) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  openedPost,
+  isLoading,
+  addCommentHandler,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [commentError, setCommentError] = useState(false);
+
+  const clearAllHandler = () => {
+    setName('');
+    setEmail('');
+    setCommentText('');
+    setNameError(false);
+    setEmailError(false);
+    setCommentError(false);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const normalizedName = name.trim();
+    const normalizedEmail = email.trim();
+    const normalizedComment = commentText.trim();
+
+    if (!normalizedName) {
+      setNameError(true);
+    }
+
+    if (!normalizedEmail) {
+      setEmailError(true);
+    }
+
+    if (!normalizedComment) {
+      setCommentError(true);
+    }
+
+    if (normalizedName && normalizedEmail && normalizedComment) {
+      setCommentError(false);
+      try {
+        await addCommentHandler(openedPost.id, normalizedName, normalizedEmail, normalizedComment);
+        setCommentText('');
```

You clear the comment text immediately after `await addCommentHandler(...)` (setCommentText('')). That assumes the handler will throw on failure. If the parent handler swallows errors (does not rethrow), this will clear the text even when the comment was not added, preventing retry. Ensure the parent rethrows on failure or change the handler to return a success indicator; clear the text only when success is confirmed. This addresses the requirement: "keep the `name` and `email` after the successful submit but clear a comment text;" and the need to allow retry on add failures (see description).